deps/v8/src/sampler.cc

```diff
@@ -342,6 +342,14 @@ void SignalHandler::HandleProfilerSignal(int signal, siginfo_t* info,
     // We require a fully initialized and entered isolate.
     return;
   }
+
+  // ISSUE-14576: To avoid deadlock, return if there is a thread lock
+  if (Isolate::GetProcessWideMutexPointer()->TryLock()) {
```

Comment:
TryLock returns false if it wasn't able to acquire lock. In this case we should return, otherwise a deadlock may happen. This is the point of the fix. The code on the next line, IsLockedByCurrentThread(), which ends up calling EnsureInitialized(), does lock, which results in a deadlock (as described in the pull request comment) and so I made sure it doesn't get there. Starting V8 version 3.29.79 the EnsureInitialized() call was removed from GetCurrentThreadId(), which fixes the deadlock problem automatically and this fix is not required anymore.

Concerning NULL pointer, it shouldn't be the case, otherwise IsLockedByCurrentThread() would also throw NULL pointer.